### DISCUSSION TITLE
Creality Sermoon D1 compatibility

### DISCUSSION THREAD

**Discussion by ratomas108** [Desuuuu/DGUS-reloaded-Klipper#33] [Q&A]
Hello,
I have tried to flash Sermoon D1 display. There is description T5L1 DGUS2 V4.0 2020.12.18 during flashing.  Original cfg file contet start with T5LC18. Is it compatible with this firmware? Is it posible to modificate this firmware with DGUS Tools?

**Comment by ratomas108** [comment]
Hello.
I have compiled source code in SW ( DGUS_Tool_V8.2.1.14 ) for T5L1 display and now it is working.

**Reply by aleksmurz** [reply]
Could you please share the T5L ported DWIN_SET directory? Can't successfully make this port on my own...

**Reply by ratomas108** [reply]
Hello,

try this.

[DWIN_SET_T5L_display.zip](https://github.com/Desuuuu/DGUS-reloaded-Klipper/files/8886543/DWIN_SET_T5L_display.zip)

**Reply by FeLiXXX81** [reply]
So this firmware is working with klipper on creality sermoon d1? 
Is it redy to flash it to screen?

**Reply by ratomas108** [reply]
Yes, this firmware is working on Creality Sermoon D1. Ready for flash to screen.

**Reply by FeLiXXX81** [reply]
Tnx for reply...I got one more question.
I flash my screen but it only show startup creality logo.
Do I have to make changes to my klipper firmware? I run it on linux laptop.

**Reply by ratomas108** [reply]
You have to use modified Klipper [https://github.com/Desuuuu/klipper/releases](url). 
![image](https://user-images.githubusercontent.com/106281051/195414464-1f5d5535-9e88-4bbb-a42b-dde4e10bb9db.png)
Section in printer.cfg
`[t5uid1]
firmware: dgus_reloaded
machine_name: Sermoon D1
volume: 30
brightness: 100
z_min: 0
z_max: 310`
I used this manual [https://3dprintbeginner.com/how-to-install-klipper-on-creality-ender-6/](url)
When you flash screen, you should see startup Klipper logo.

**Reply by ahagie** [reply]
Can someone please put a link tot he correct .bin file for the Sermoon D1 on Klipper. I see a few in the Files list. Just need 1 the Works WITH BL Touch please (not a programmer here, looking to flash eprom and replace cfg on sonic pad or pi for the Sermoon.)

**Reply by FeLiXXX81** [reply]
There is no .bin file for Sermoon-D1 firmware. You can only download configuration file for Sermoon-D1. 
After you install klipper you have to create it and then flash your mainboard with created firmware.bin file.
So go and watch how to install klipper for Sermoon-D1 on YouTube and you will see how to do it. 
It is 5 minut job when you know how to.

**Reply by clmnsr** [reply]
Can you tell me, how you compile firmware for the Sermoon D1?
I use the Klipper repository from Desuuu. These are my settings:
![image](https://user-images.githubusercontent.com/60857749/218989694-17923b12-4c2c-45e6-9290-faedf52b0dd7.png)

**Comment by ahagie** [comment]
But if I have a stock Sermoon d1. The same bin gil would be for all Sermoon d1 printers correct? Then I could just bring up pi or the Creality  pad to copy the printer cuff file. Wouldn’t that be the same for ever d1?

On Oct 19, 2022, at 10:41 AM, FeLiXXX81 ***@***.***> wrote:

﻿

There is no .bin file for Sermoon-D1 firmware. You can only download configuration file for Sermoon-D1.
After you install klipper you have to create it and then flash your mainboard with created firmware.bin file.
So go and watch how to install klipper for Sermoon-D1 on YouTube and you will see how to do it.
It is 5 minut job when you know how to.

**Reply by FeLiXXX81** [reply]
Look anyway you have to install klipper. So when you have it on pi you can make your firmware file. I dont know if it will work if I give you my file. I have klipper on laptop not on pi.
